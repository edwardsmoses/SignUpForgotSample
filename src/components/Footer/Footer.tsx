import React from 'react';
import { StyleSheet, TouchableOpacity, Text, View } from 'react-native';

import * as Colors from '../../styles/Color';

interface FooterProps {
    onPress?: () => {};
    content?: string;
}

const Footer: React.FC<FooterProps> = ({
    content = 'Already have an account? Sign In',
    onPress,
}) => {
    return (
        <View style={styles.containerStyle}>
            <View style={styles.dividerStyle} />
            <TouchableOpacity onPress={onPress} style={styles.footerStyle}>
                <Text style={styles.textStyle}>{content}</Text>
            </TouchableOpacity>
        </View>
    );
};

export default Footer;

const styles = StyleSheet.create({
    containerStyle: {
        alignContent: 'center',
        justifyContent: 'center',
    },
    dividerStyle: {
        borderTopColor: Colors.Blue,
        borderTopWidth: StyleSheet.hairlineWidth,
        marginBottom: 15,
        marginTop: 10,
        alignSelf: 'stretch',
    },
    footerStyle: {
        alignContent: 'center',
        justifyContent: 'center',
    },
    textStyle: {
        color: Colors.Blue,
        fontSize: 20,
        textAlign: 'center',
    },
});
