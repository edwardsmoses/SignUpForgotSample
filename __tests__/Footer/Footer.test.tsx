import { render, cleanup } from 'react-native-testing-library';
import React from 'react';
import Footer from '../../src/components/Footer/Footer';

afterEach(cleanup);

describe('<Footer />', () => {
    it('should match snapshot', () => {
        const rendered = render(
            <Footer
                onPress={() => {
                    return {};
                }}
                content="Already have an account?"
            />,
        ).toJSON();

        expect(rendered).toMatchSnapshot();
    });
});
