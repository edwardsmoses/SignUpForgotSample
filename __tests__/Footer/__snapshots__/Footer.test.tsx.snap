// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should match snapshot 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "borderTopColor": "#53a1e8",
        "borderTopWidth": 0.5,
        "marginBottom": 15,
        "marginTop": 10,
      }
    }
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignContent": "center",
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#53a1e8",
          "fontSize": 20,
          "textAlign": "center",
        }
      }
    >
      Already have an account?
    </Text>
  </View>
</View>
`;
